src/Streamly/Internal/LZ4.hs

```diff
@@ -130,6 +136,49 @@ lz4_MAX_OUTPUT_SIZE :: CInt
 lz4_MAX_OUTPUT_SIZE =
     min (unsafePerformIO $ c_compressBound lz4_MAX_INPUT_SIZE) maxBound
 
+--------------------------------------------------------------------------------
+-- Conversion helpers
+--------------------------------------------------------------------------------
+
+{-# INLINE cIntToInt #-}
+cIntToInt :: CInt -> Int
+cIntToInt = fromIntegral
+
+{-# INLINE intToCInt #-}
+intToCInt :: Int -> CInt
+intToCInt = fromIntegral
+
+{-# INLINE i32ToInt #-}
+i32ToInt :: Int32 -> Int
+i32ToInt = fromIntegral
+
+{-# INLINE cIntToI32 #-}
+cIntToI32 :: CInt -> Int32
+cIntToI32 = coerce
+
+{-# INLINE i32ToCInt #-}
+i32ToCInt :: Int32 -> CInt
+i32ToCInt = coerce
+
+--------------------------------------------------------------------------------
+-- Configuration
+--------------------------------------------------------------------------------
+
+data Config =
+    Config
+        { maxUncompressedSize :: Int
+        , encodeUncompressedSize :: Bool
+        }
+
+defaultConfig :: Config
+defaultConfig = Config (2 * 1024 * 1024 * 1024 - 64) True
+
+setMaxUncompressedSize :: Config -> Int -> Config
+setMaxUncompressedSize c i = c { maxUncompressedSize = i }
```

Comment:
Swap the order of the arguments so that we can get a `Config -> Config` function which we can use in our APIs without having to pass a `Config` argument.